Realm/RLMHandover.h

```diff
@@ -0,0 +1,67 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+@class RLMRealm;
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// A Realm-bound object that can only be passed between threads by exporting for handover
```

Comment:
An object which is bound to a thread-specific `RLMRealm` instance, and so cannot be passed between threads.

Objects of classes conforming to this protocol can be packaged for transport between threads by calling `-[RLMRealm exportThreadHandoverWithObjects:]`. Note that only types defined by Realm can meaningfully conform to this protocol, and defining new classes which attempt to conform to it will not make them work with `exportThreadHandoverWithObjects`.